RecoLocalCalo/HGCalRecProducers/python/hgcalLayerClusters_cff.py

```diff
@@ -6,30 +6,36 @@
 
 from RecoLocalCalo.HGCalRecProducers.HGCalUncalibRecHit_cfi import HGCalUncalibRecHit
 
-from SimCalorimetry.HGCalSimProducers.hgcalDigitizer_cfi import fC_per_ele, hgceeDigitizer, hgchebackDigitizer, hfnoseDigitizer
+from SimCalorimetry.HGCalSimProducers.hgcalDigitizer_cfi import fC_per_ele, HGCAL_noises, hgceeDigitizer, hgchebackDigitizer, hfnoseDigitizer
 
-hgcalLayerClusters = hgcalLayerClusters_.clone()
-
-hgcalLayerClusters.timeOffset = hgceeDigitizer.tofDelay
-hgcalLayerClusters.plugin.dEdXweights = cms.vdouble(dEdX.weights)
-hgcalLayerClusters.plugin.fcPerMip = cms.vdouble(HGCalUncalibRecHit.HGCEEConfig.fCPerMIP + HGCalUncalibRecHit.HGCHEFConfig.fCPerMIP)
-hgcalLayerClusters.plugin.thicknessCorrection = cms.vdouble(HGCalRecHit.thicknessCorrection)
-hgcalLayerClusters.plugin.sciThicknessCorrection = HGCalRecHit.sciThicknessCorrection
-hgcalLayerClusters.plugin.deltasi_index_regemfac = HGCalRecHit.deltasi_index_regemfac
-hgcalLayerClusters.plugin.fcPerEle = cms.double(fC_per_ele)
-hgcalLayerClusters.plugin.noises = cms.PSet(refToPSet_ = cms.string('HGCAL_noises'))
-hgcalLayerClusters.plugin.noiseMip = hgchebackDigitizer.digiCfg.noise
+hgcalLayerClusters = hgcalLayerClusters_.clone(
+    timeOffset = hgceeDigitizer.tofDelay,
+    plugin = dict(
+        dEdXweights = dEdX.weights,
```

Comment:
I tend to forget that assignment statements in Python do not copy objects (as in C++) but they create bindings between a target and an object. Therefore, if we assign (e.g.)
```
        dEdXweights = dEdX.weights,
```
and then modiffy `dEdXweights` afterwards, we are going to modify also the source `dEdX.weight` and all other configs which use it.

To avoid it one can make a deep copy, or assign the value of the object copied from (which was by the way what coded in the original suggestion of @slava77).

In this case one could therefore assign
```
        dEdXweights = dEdX.weights.value(),
```
and the same should also get applied to lines 17, 18, 19, 20, 23, 24 below, as well as in the corresponding lines within `hgcalLayerClustersHFNose`.

(As of now, none of those parameters get modified in our configs, and therefore there will be no practical effect on the actual configs. Let consider it as a safe practice to avoid possible future issues, in this case)

Reply:
thanks a lot for the detailed explanation!